curriculum/challenges/english/25-front-end-development/review-typescript/6724e296dceca21b82426229.md

```diff
@@ -9,6 +9,238 @@ dashedName: review-typescript
+- **Compiler**: You first need to compile TypeScript code into regular JavaScript. When you run the compiler, TypeScript will evaluate your code and throw an error for any issues where the types don't match.
+
+For example, you can define a type for the `array` parameter as follows:
+
+```typescript
+const getRandomValue = (array: string[]) => {
+  return array[Math.floor(Math.random() * array.length)];
+}
+```
+
+This type definition tells TypeScript that the `array` argument must be an array of strings. Then, when you call `getRandomValue` and pass it a number, you get an error called a compiler error.
+
+## Data Types in TypeScript
+
+- **Primitive Data Types in TypeScript**: For the primitive data types `string`, `null`, `undefined`, `number`, `boolean`, and `bigint`, TypeScript offers corresponding type keywords.
+
+```typescript
+let str: string = "Naomi";
+let num: number = 42;
+let bool: boolean = true;
+let nope: null = null;
+let nada: undefined = undefined;
+```
+
+- **Array**: You can define an array of specific type with two different syntaxes.
+
+```typescript
+const arrOne: string[] = ["Naomi"];
+const arrTwo: Array<string> = ["Camperchan"];
+```
+
+- **Objects**: You can define the exact structure of an object.
+
+```typescript
+const obj: { a: string, b: number } = { a: "Naomi", b: 42 };
+```
+
+If you want an object with any keys, but where all values must be strings, there are two ways to define it:
+
+```typescript
+const objOne: Record<string, string> = {};
+const objTwo: { [key: string]: string } = {};
+```
+
+- **Other Helpful Types in TypeScript**:
+  - **`any`**: `any` indicates that a value can have any type. It tells the compiler to stop caring about the type of that variable.
+  - **`unknown`**: `unknown` tells TypeScript that you _do_ care about the type of the value, but you don’t actually know what it is. `unknown` is generally preferred over `any`.
+  - **`void`**: This is a special type that you’ll typically only use when defining functions. Functions that don’t have a return value use a return type of `void`.
+  - **`never`**: It represents a type that will never exist.
+
+- **`type` Keyword**: This keyword is like `const`, but instead of declaring a variable, you can declare a type.
+
+It is useful for declaring custom types such as union types or types that include only specific values:
+
+```typescript
+type stringOrNumber = string | number;
+type bot = "camperchan" | "camperbot" | "naomi";
+```
+
+- **`interface`**: Interfaces are like classes for types. They can implement or extend other interfaces, are specifically object types, and are generally preferred unless you need a specific feature offered by a `type` declaration.
+
+```typescript
+interface wowie {
+  zowie: boolean;
+  method: () => void;
+}
+```
+
+- **Defining Return Type**: You can also define the _return type_ of the function.
```

Comment:
let's add a sentence here explaining that in the example, the function return must be a a string otherwise TypeScript will show an error message.